enos/enos.py

```diff
@@ -53,23 +58,52 @@
 
 
 @enostask("""
-usage: enos up [-f CONFIG_PATH] [--force-deploy] [-t TAGS | --tags=TAGS]
-               [--provider=PROVIDER] [-vv|-s|--silent]
+usage: enos up  [-e ENV|--env=ENV][-f CONFIG_PATH] [--force-deploy]
+                [--provider=PROVIDER] [-t TAGS|--tags=TAGS]
+                [-v|-vv|-s|--silent]
 
 Get resources and install the docker registry.
 
 Options:
-  -h --help            Show this help message.
+  -e ENV --env=ENV     Path to the environment directory. You should
+                       use this option when you want to link to a specific
+                       experiment. Do not specify it in other cases.
   -f CONFIG_PATH       Path to the configuration file describing the
                        deployment [default: ./reservation.yaml].
+  -h --help            Show this help message.
   --force-deploy       Force deployment [default: False].
-  -t TAGS --tags=TAGS  Only run ansible tasks tagged with these values.
   --provider=PROVIDER  The provider name [default: G5k].
+  -t TAGS --tags=TAGS  Only run ansible tasks tagged with these values.
+  -v --version         Show version number.
+  -vv                  Verbose mode.
+  -s --silent          Quiet mode.
 
 """)
 def up(provider=None, env=None, **kwargs):
     logging.debug('phase[up]: args=%s' % kwargs)
 
+    # Generates a directory for results
+    resultdir_name = 'enos_' + datetime.today().isoformat()
+    if kwargs['--env']:
+        resultdir_name = kwargs['--env']
+    resultdir = os.path.join(CALL_PATH, resultdir_name)
+    try:
+        os.mkdir(resultdir)
+    except OSError:
+        pass
```

Comment:
Before passing, I recommend to add a warning message: `logging.warn("%s directory already exist" % resultdir)`